libr/anal/var.c

```diff
@@ -262,35 +259,59 @@ R_API RAnalVar *r_anal_function_get_var(RAnalFunction *fcn, char kind, int delta
 	return NULL;
 }
 
-// TODO: This should be just r_anal_var_addr(RAnalVar *) without querying by name
-R_API ut64 r_anal_var_addr(RAnal *a, RAnalFunction *fcn, const char *name) {
+R_API ut64 r_anal_var_addr(RAnalVar *var) {
+	r_return_val_if_fail (var, UT64_MAX);
+	RAnal *anal = var->fcn->anal;
 	const char *regname = NULL;
-	ut64 ret = UT64_MAX;
-	if (!a || !fcn) {
-		return ret;
+	if (var->kind == R_ANAL_VAR_KIND_BPV) {
+		regname = r_reg_get_name (anal->reg, R_REG_NAME_BP);
+	} else if (var->kind == R_ANAL_VAR_KIND_SPV) {
+		regname = r_reg_get_name (anal->reg, R_REG_NAME_SP);
 	}
-	RAnalVar *v1 = r_anal_function_get_var_byname (fcn, name);
-	if (v1) {
-		if (v1->kind == R_ANAL_VAR_KIND_BPV) {
-			regname = r_reg_get_name (a->reg, R_REG_NAME_BP);
-		} else if (v1->kind == R_ANAL_VAR_KIND_SPV) {
-			regname = r_reg_get_name (a->reg, R_REG_NAME_SP);
+	return r_reg_getv (anal->reg, regname) + var->delta;
+}
+
+R_API st64 r_anal_function_get_var_stackptr_at(RAnalFunction *fcn, int delta, ut64 addr) {
+	st64 offset = (st64)addr - (st64)fcn->addr;
```

Comment:
looks wrong. that wont work when the higher bit is set

Reply:
It seems to work fine. I added tests for exactly these cases.